src/main/scala/org/hammerlab/guacamole/loci/SimpleRange.scala

```diff
@@ -0,0 +1,37 @@
+package org.hammerlab.guacamole.loci
+
+import java.lang.{Long => JLong}
+
+import com.google.common.collect.{Range => JRange}
+
+import scala.collection.Iterator
+
+/**
+ * A range of Longs. Inclusive on start, exclusive on end.
+ */
+case class SimpleRange(start: Long, end: Long) extends Ordered[SimpleRange] {
+  /** Iterate through elements in the range. */
+  def iterator(): Iterator[Long] = new Iterator[Long] {
+    private var i = start
+    override def hasNext: Boolean = i < end
+    override def next(): Long =
+      if (hasNext) { val result = i; i += 1; result }
+      else Iterator.empty.next()
+  }
+  /** Number of elements in the range. */
+  def length: Long = end - start
+
+  /** Comparisons between ranges. Order is DESCENDING (i.e. reversed) from by start. */
+  def compare(other: SimpleRange): Int = {
```

Comment:
It actually doesn't when it's overriding a `trait` method, but it's good to add on balance, good call; done!